fc/fc_app.py

```diff
@@ -219,124 +231,162 @@ def estimate_job_size(num_tasks):
+        qsub_params=dict(
+            mem='4G',
+            wd=True,
+            ncpus=1,
+            walltime='1h',
+            name='fc-generate-{}'.format(tag)
+        )
+    )
 
 
 @cli.command(help='Generate Tasks into file and Queue PBS job to process them')
-@click.option('--project', '-P', default='u46')
-@click.option('--queue', '-q', default='normal',
-              type=click.Choice(['normal', 'express']))
-@click.option('--year', 'time_range', callback=task_app.validate_year, help='Limit the process to a particular year')
-@click.option('--no-qsub', 'no_qsub', is_flag=True, default=False, help="Skip submitting qsub for next step")
+@click.option('--no-qsub', is_flag=True, default=False, help="Skip submitting qsub for next step")
+@click.option(
+    '--task-desc', 'task_desc_file', help='Task environment description file',
+    required=True,
+    type=click.Path(exists=True, readable=True, writable=False, dir_okay=False)
+)
 @tag_option
-@task_app.app_config_option
-@task_app.save_tasks_option
-@ui.config_option
 @ui.verbose_option
 @ui.log_queries_option
 @ui.pass_index(app_name=APP_NAME)
-def generate(index,
-             app_config,
-             project,
-             queue,
-             output_tasks_file,
-             time_range,
-             no_qsub,
-             tag):
-    output_tasks_file = Path(output_tasks_file).absolute()
-
-    config, tasks = task_app.load_config(index, app_config, make_fc_config, make_fc_tasks, time_range=time_range)
-
-    num_tasks_saved = task_app.save_tasks(config, tasks, output_tasks_file)
+def generate(index: Index,
+             task_desc_file: str,
+             no_qsub: bool,
+             tag: str):
     _LOG.info('Tag: %s', tag)
+
+    config, task_desc = _make_config_and_description(index, Path(task_desc_file))
+
+    num_tasks_saved = task_app.save_tasks(
+        config,
+        make_fc_tasks(index, config, query=task_desc.parameters.query),
+        task_desc.runtime_state.task_serialisation_path
+    )
     _LOG.info('Found %d tasks', num_tasks_saved)
 
-    nodes, walltime = estimate_job_size(num_tasks_saved)
+    if not num_tasks_saved:
+        _LOG.info("No tasks. Finishing.")
+        sys.exit(0)
 
+    nodes, walltime = estimate_job_size(num_tasks_saved)
     _LOG.info('Will request %d nodes and %s time', nodes, walltime)
 
     if no_qsub:
-        _LOG.info('Quitting early as requested')
+        _LOG.info('Skipping submission due to --no-qsub')
         return 0
 
-    qsub = QSubLauncher(norm_qsub_params(
-        {'project': project,
-         'queue': queue,
-         'name': 'fc-run-{}'.format(tag),
-         'mem': 'small',
-         'wd': True,
-         'noask': True,
-         'nodes': nodes,
-         'walltime': walltime,
-         'extra_qsub_args': ['-W', 'umask=027']}))
+    submit_subjob(
+        name='run',
+        task_desc=task_desc,
+
+        command=[
+            'run',
+            '-vv',
+            '--task-desc', str(task_desc_file),
```

Comment:
AHA! I'd missed this when looking at https://github.com/GeoscienceAustralia/digitalearthau/pull/32

The vital linking/missing piece.